esp/esp/program/modules/handlers/programprintables.py

```diff
@@ -788,43 +789,87 @@ def getSchedule(program, user, schedule_type=None, room_numbers=True):
                 classes = ProgramPrintables.get_teacher_classlist(program, user)
+
+            schedule = """
+                        <p> %s schedule for %s: </p>
+                        """ % (schedule_type, user.name())
+            schedule += """
+                        <table cellspacing="0" cellpadding="10" border="1">
+                            <tr>
+                            <th width="100">Time</th>
+                            <th width="300">Class</th>
+                            """
+            if room_numbers:
+                schedule += """
+                            <th width="100">Room</th>
+                            """
+            schedule += """
+                        </tr>
+                        <indent>
+                        """
             for cls in classes:
-                rooms = cls.prettyrooms()
-                if len(rooms) == 0:
-                    rooms = u' N/A'
+                times = cls.friendly_times()
+                if len(times) == 0:
+                    # don't show classes with no times
+                    continue
                 else:
-                    rooms = u' ' + u", ".join(rooms)
+                    times = ' ' + ', '.join(times)
+                schedule += """
+                            <tr>
+                            <td> %s </td>
+                            <td> %s </td>
+                            """ % (str(times), cls.title())
                 if room_numbers:
-                    schedule += u'%s|%s|%s\n' % ((u' '+u",".join(cls.friendly_times())).ljust(24), (u' ' + cls.title()).ljust(40), rooms)
-                else:
-                    schedule += u'%s|%s\n' % ((u' '+u",".join(cls.friendly_times())).ljust(24), (u' ' + cls.title()).ljust(40))
+                    rooms = cls.prettyrooms()
+                    if len(rooms) == 0:
+                        rooms = 'N/A'
+                    else:
+                        rooms = ' ' + ', '.join(rooms)
+                    schedule += """
+                                <td> %s </td>
+                                """ % (str(rooms))
+                schedule += """
+                            </tr>
+                            """
+            schedule += """
+                        </indent>
+                        </table>
+                        """
 
         elif schedule_type == u'Volunteer':
-            schedule = u"""
-Volunteer schedule for %s:
-
- Time                   | Shift                                  \n""" % (user.name())
-            schedule += u'------------------------+----------------------------------------\n'
+            schedule = """
+                       <p> %s schedule for %s: </p>
+                       """ % (schedule_type, user.name())
+            schedule += """
+                        <table cellspacing="0" cellpadding="10" border="1">
+                            <tr>
+                            <th width="100">Time</th>
```

Comment:
You might want to use percentages here instead of pixels, otherwise the width of the email could be way too big, especially on mobile

Reply:
(And above for students/teachers)